web/tests/api/v2/verify.test.ts

```diff
@@ -0,0 +1,508 @@
+import { POST } from "@/api/v2/verify";
+import { VerificationLevel } from "@worldcoin/idkit-core";
+import { NextRequest } from "next/server";
+import { semaphoreProofParamsMock } from "../__mocks__/proof.mock";
+
+// #region Mocks
+const FetchAppAction = jest.fn();
+const InsertNullifier = jest.fn();
+const UpdateNullifierUses = jest.fn();
+
+jest.mock(
+  "../../../api/v2/verify/graphql/fetch-app-action.generated",
+  jest.fn(() => ({
+    getSdk: () => ({
+      FetchAppAction,
+    }),
+  })),
+);
+
+jest.mock(
+  "../../../api/v2/verify/graphql/insert-nullifier.generated",
+  jest.fn(() => ({
+    getSdk: () => ({
+      InsertNullifier,
+    }),
+  })),
+);
+
+jest.mock(
+  "../../../api/v2/verify/graphql/update-nullifier-uses.generated",
+  jest.fn(() => ({
+    getSdk: () => ({
+      UpdateNullifierUses,
+    }),
+  })),
+);
+jest.mock("../../../lib/logger", () => ({
+  logger: {
+    error: jest.fn(),
+    warn: jest.fn(),
+  },
+}));
+
+jest.mock("../../../services/posthogClient", () => ({
+  captureEvent: jest.fn(),
+}));
+
+global.fetch = jest.fn(() =>
+  Promise.resolve({
+    json: jest.fn(() => Promise.resolve({})),
+    ok: true,
+    status: 200,
+  }),
+) as jest.Mock;
+
+const mockFetch = (params: {
+  body: Record<string, any>;
+  ok: boolean;
+  status: number;
+  text?: string;
+}) => {
+  (fetch as jest.Mock).mockImplementationOnce(() =>
+    Promise.resolve({
+      json: jest.fn(() => Promise.resolve(params.body)),
+      text: jest.fn(() => Promise.resolve(params.text ?? "")),
+      ok: params.ok,
+      status: params.status,
+    }),
+  );
+};
+// #endregion
+
+// #region Test data
+const getUrl = (app_id: string) =>
+  new URL(`/api/v2/verify/${app_id}`, "http://localhost:3000");
+
+const validBody = {
+  ...semaphoreProofParamsMock,
+  action: "verify",
+};
```

Comment:
unhashed signal shouldn't be included

```suggestion
const validBody = {
  ...semaphoreProofParamsMock,
  signal: undefined,
  action: "verify",
};
```

Reply:
Updated